apps/privacy-panel/js/tc/applications.js

```diff
@@ -123,6 +132,36 @@ function(panels, appList, appDetails) {
       });
 
       _appListContainer.appendChild(list);
+    },
+
+    /**
+     * Filter the application list when the 'search' mode is on.
+     *
+     * @method filterAppList
+     * @param {String} pattern  Search pattern.
+     */
+    filterAppList: function filterAppList() {
+      var pattern = _appSearchInput.value.replace(/^\s+|\s+$/g, '');
+      var re = pattern && pattern.length ? new RegExp(pattern, 'i') : null;
+      for (var i = 0, l = _appItems.length; i < l; i++) {
+        _appItems[i].hidden = re && !re.test(_appItems[i].dataset.key);
+      }
+    },
+
```

Comment:
that said, I'd like to exit search mode when I select an app. otherwise when coming back to the test panel the list will still be filtered but the "close" button won't be there anymore.